### PR TITLE
Fix report step on CI

### PR DESCRIPTION
**Pull request by hendrikmakait**

### CODE REVIEW

File: tests/tpch/conftest.py [hunk 1]
```diff
@@ -30,9 +30,6 @@ def pytest_addoption(parser):
     parser.addoption("--cloud", action="store_false", dest="local", help="")
     parser.addoption("--restart", action="store_true", default=True, help="")
     parser.addoption("--no-restart", action="store_false", dest="restart", help="")
-    parser.addoption(
-        "--performance-report", action="store_true", default=False, help=""
-    )
     parser.addoption(
```
Comment (jrbourbeau) [inline tests/tpch/conftest.py:33]: Not clear to me why this is being removed, but that's okay. Just wanted to make sure this is intentional is all

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -262,7 +262,7 @@ jobs:
           script: |
             const workflow_url = `https://github.com/${process.env.GITHUB_REPOSITORY}/actions/runs/${process.env.GITHUB_RUN_ID}`
             const issue_body = `[Workflow Run URL](${workflow_url})`
-            github.issues.create({
+            github.rest.issues.create({
                 owner: context.repo.owner,
                 repo: context.repo.repo,
                 body: issue_body,
```
Comment (jrbourbeau) [inline .github/workflows/tests.yml:268]: Don't need to do anything here, but just wanted to note that the original version of this code has been moved into it's own GitHub action https://github.com/xarray-contrib/issue-from-pytest-log which is what we're now using in `dask/dask`

https://github.com/dask/dask/blob/98331028f8a51d404f8c9856d49ba726322c4e03/.github/workflows/upstream.yml#L76-L86

Comment (jrbourbeau) [reply .github/workflows/tests.yml:268]: To be clear, this is mostly just an FYI

File: tests/tpch/conftest.py (hunk 1 above)
Comment (hendrikmakait) [reply tests/tpch/conftest.py:33]: This PR included https://github.com/coiled/benchmarks/pull/1599.

**Pull request merged**